core/src/main/java/org/eclipse/hono/connection/ConnectionFactory.java

```diff
@@ -88,4 +93,168 @@ void connect(
             Handler<AsyncResult<ProtonConnection>> closeHandler,
             Handler<ProtonConnection> disconnectHandler,
             Handler<AsyncResult<ProtonConnection>> connectionResultHandler);
+
+    /**
+     * Builder for ConnectionFactory instances.
+     */
+    final class ConnectionFactoryBuilder {
```

Comment:
FMPOV we should rather get rid of the builder altogether since we do not really use it anyway. People usually create a `HonoClient` using the constructor that accepts a vert.x and `ClientConfigProperties` instance which creates the connection factory under the hood. This builder has been around since the beginning but turned out to be superfluous FMPOV.

Reply:
See above. We could collapse both builders into one.